message_parser/src/main/java/cumulus_message_adapter/message_parser/MessageAdapter.java

```diff
@@ -0,0 +1,127 @@
+package cumulus_message_adapter.message_parser;
+
+import com.amazonaws.services.lambda.runtime.Context; 
+
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.util.Scanner;
+import java.util.Map;
+import java.util.HashMap;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+public class MessageAdapter implements IMessageAdapter
+{
+    /**
+     * Call to message adapter zip to execute a message adapter function. Pass args through the process input
+     * and read return result from process output.
+     * @param messageAdapterFunction - 'loadRemoteEvent', 'loadNestedEvent', or 'createNextEvent'
+     * @param inputJson - argument to message adapter function. Json that contains all of the params.
+     * @return the return from the message adapter function
+     */
+    public String CallMessageAdapterFunction(String messageAdapterFunction, String inputJson)
+        throws MessageAdapterException
+    {
+        String messageAdapterOutput = "";
+
+        try
+        {
+            ProcessBuilder processBuilder = new ProcessBuilder("python", "cumulus-message-adapter.zip", messageAdapterFunction);
+
+            Process process = processBuilder.start();
+
+            OutputStreamWriter writer = new OutputStreamWriter(process.getOutputStream());
+            writer.write(inputJson);
+            writer.close();
+
+            // Log the entire error
+            // TO DO: Update logging
+            Scanner scanner = new Scanner(process.getErrorStream());
+            Boolean hasError = false;
+            while(scanner.hasNextLine()) 
+            {
+                hasError = true;
+                System.out.println(String.format("Cumulus Message Adapter error: %s: %s", messageAdapterFunction, scanner.nextLine()));  
+            }
+            scanner.close();
+
+            if(hasError)
+            {
+                throw new MessageAdapterException("Error executing " + messageAdapterFunction);
+            }
+
+            scanner = new Scanner(process.getInputStream());
+            if(scanner.hasNextLine()) 
+            {
+                messageAdapterOutput = scanner.nextLine();
+            }
+            scanner.close();
+        }
+        catch(IOException e)
+        {
+            throw new MessageAdapterException("Unable to find Cumulus Message Adapter", e.getCause());      
+        }
```

Comment:
What happens to other types of exceptions that could be thrown?

Reply:
If the method throws an exception, Java forces you to do something with it. If there was another exception type able to be thrown here, I'd have to handle it.